src/engine/renderer/tr_image_ktx.cpp

```diff
@@ -179,18 +179,17 @@ bool LoadInMemoryKTX( const char *name, void *ktxData, size_t ktxSize,
 		IsNonArrayCubemapTexture(hdr) ? hdr->numberOfFaces : 1U};
 
 	for(uint32_t i{ 0 }; i < hdr->numberOfMipmapLevels; i++) {
+		if ( !IsValidKTXFileStreamPosition( ptr, ktxSize, static_cast<const byte*>(ktxData) ) ) {
+			Log::Warn("KTX image '%s' has bad header or texture data", name);
+			return false;
+		}
+
 		const uint32_t imageSize{ GetImageSize( ptr, needReverseBytes ) };
 
 		totalImageSize += imageSize * mipmapLevelCoefficient;
 		ptr += sizeof(uint32_t) + imageSize;
 	}
 
-	// ptr points to next byte after ktxData buffer end.
```

Comment:
The check at the end is still needed isn't it?

Reply:
@slipher: can you write review comments _under_ the line you want to comment? The GitHub workflow assumes review comments to be done under the line? When review comments are done above the line, the GitHub UI hides the line you reviewed, making things hard to read, requiring manual action to workaround that and understand what's going on…

Reply:
@slipher Correct, will add last check.

Reply:
Did you forget to push it? I don't see any new changes.

Reply:
Done